lib/spack/spack/main.py

```diff
@@ -293,7 +295,18 @@ def add_subcommand_group(title, commands):
     def add_subparsers(self, **kwargs):
         """Ensure that sensible defaults are propagated to subparsers"""
         kwargs.setdefault('metavar', 'SUBCOMMAND')
+
+        # From Python 3.7 we can require a subparser, earlier versions
+        # of argparse will error because required=True is unknown
```

Comment:
This is the only PR I ever submitted to cpython: https://github.com/python/cpython/pull/16588